src/libraries/Common/src/Interop/Windows/Kernel32/Interop.Globalization.cs

```diff
@@ -133,6 +138,12 @@ internal static unsafe partial class Kernel32
         [DllImport("kernel32.dll", CharSet = CharSet.Unicode)]
         internal static extern int GetCalendarInfoEx(string? lpLocaleName, uint Calendar, IntPtr lpReserved, uint CalType, IntPtr lpCalData, int cchData, IntPtr lpValue);
 
+        [DllImport("kernel32.dll")]
+        internal static extern int GetUserGeoID(int geoClass);
+
+        [DllImport("kernel32.dll", CharSet = CharSet.Unicode)]
+        internal static extern int GetGeoInfo(int location, int geoType, ref char lpGeoData, int cchData, int LangId);
```

Comment:
```suggestion
        internal static extern int GetGeoInfo(int location, int geoType, char* lpGeoData, int cchData, int LangId);
```

Reply:
I had this in the first place and changed when you asked to use Span. otherwise the calling code will not be nice. I think either we keep using char* and not span in the calling code, or keeping the ref here. otherwise, I have to use other options like fixed or use Unsafe.AsPointer I think which I am not seeing it is worth doing that in this simple case.

Reply:
What about something like this?
```
int geoIsoIdLength;

fixed (char* pGeoIso2Letters = geoIso2Letters)
{
    geoIsoIdLength = Interop.Kernel32.GetGeoInfo(geoId, Interop.Kernel32.GEO_ISO2, pGeoIsoIdLength, geoIso2Letters.Length, Interop.Kernel32.EN_LANG_ID);
}

if (geoIsoIdLength ...
```

Reply:
> I have to use other options like fixed

I do not see a problem with using fixed around an interop call. We do that in thousand other places libraries.

Reply:
I don't like using fixed with stackalloc allocated buffer. if believe we should do it this way, I wouldn't mind.

Reply:
What's the benefit here of stackalloc'ing into a span rather than into a ptr if we then turn around and immediately use fixed to get the pointer and do basically nothing else with the span?

Reply:
Span gives you bounds checks for the string manipulations. I do not want do have another security even where one-time globalization initialization leads to buffer overrun.

Reply:
> Span gives you bounds checks for the string manipulations.

The only string manipulation here is the subsequent Slice, and we're feeding into it the result of the GetGeoInfo call.  If GetGeoInfo was buggy and gave us a bad length, it could be buggy in other ways, such as having written all the way up to that bad length.

But if you think this adds meaningful safety, ok.

Reply:
I'll do what Jan suggested then and use `fixed` in the calling code.